atomic_reactor/schemas/container.json

```diff
@@ -156,6 +156,51 @@
       "description": "build-step plugin to be used",
       "enum": ["docker_api", "imagebuilder", "buildah_bud"]
     },
+    "remote_source": {
```

Comment:
Wouldn't be better to having it in this format for future extension:

```
remote_sources: [
    {
      "repo": "git.example.com/proj-go",
      "ref": "b55c00f45ec3dfee0c766cea3d395d6e21cc2e5a",
      "pkg_manager": "gomod",
      "flags": [...cachito flags...],
   }, {
      "repo": "another.git.com/pyproject",
      "ref": "651abc41170e52866148cc0c99c3671a2ace7002",
      "pkg_manager": "pip",
      "flags": [...cachito flags...],
   }
]
```

Reply:
During the design phase, we decided to only support a single remote source. But a single remote source could have multiple package managers. This is inline with upstream projects.

Having multiple remote sources is challenging because if you have, for example, two goloang apps, the value for GOCACHE/GOPATCH is ambiguous. 

Given the complexities in implementing this, we decided that this could be added later on if there was need.

Reply:
During the design phase we decided to only support a single remote source, but allow multiple package managers for it. 

Supporting multiple remote sources can be challenging. For instance, if there are two remote sources using gomod, it's unclear what the value of GOCACHE/GOPATH should be set to. Should it point to the cache of the first remote source, or the second?

Given the complexities that can arise from this scenario, we decided to only support a single remote source until a real use case arises to makes us reconsider.

Reply:
ok